### PR TITLE
[Nougat] Fix pipeline

### LINKED ISSUE #27475: [nougat] Unable to use nougat models with `image-to-text` pipeline
### System Info

- `transformers` version: 4.36.0.dev0
- Platform: Linux-5.15.120+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.17.3
- Safetensors version: 0.4.0
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.1.0+cu118 (False)
- Tensorflow version (GPU?): 2.14.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.7.5 (cpu)
- Jax version: 0.4.20
- JaxLib version: 0.4.20
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@NielsRogge @Narsil

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Running
```py
from transformers import pipeline
pipe = pipeline('image-to-text', 'facebook/nougat-base')
pipe('https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/nougat_paper.png')
```

results in the following error:
```
ValueError: Unrecognized feature extractor in facebook/nougat-base. Should have a `feature_extractor_type` key in its preprocessor_config.json of config.json, or one of the following `model_type` keys in its config.json: audio-spectrogram-transformer, beit, chinese_clip, clap, clip, clipseg, clvp, conditional_detr, convnext, cvt, data2vec-audio, data2vec-vision, deformable_detr, deit, detr, dinat, donut-swin, dpt, encodec, flava, glpn, groupvit, hubert, imagegpt, layoutlmv2, layoutlmv3, levit, maskformer, mctct, mobilenet_v1, mobilenet_v2, mobilevit, nat, owlvit, perceiver, poolformer, pop2piano, regnet, resnet, seamless_m4t, segformer, sew, sew-d, speech_to_text, speecht5, swiftformer, swin, swinv2, table-transformer, timesformer, tvlt, unispeech, unispeech-sat, van, videomae, vilt, vit, vit_mae, vit_msn, wav2vec2, wav2vec2-conformer, wavlm, whisper, xclip, yolos
```

### Expected behavior

The model should function properly with the `pipeline` API.

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR ensures [Nougat](https://huggingface.co/docs/transformers/main/model_doc/nougat) works with the image-to-text pipeline.

Fixes #27475

### CODE REVIEW

File: src/transformers/pipelines/__init__.py [hunk 1]
```diff
@@ -415,7 +415,8 @@
 # any tokenizer/feature_extractor might be use for a given model so we cannot
 # use the statically defined TOKENIZER_MAPPING and FEATURE_EXTRACTOR_MAPPING to
 # see if the model defines such objects or not.
-MULTI_MODEL_CONFIGS = {"SpeechEncoderDecoderConfig", "VisionEncoderDecoderConfig", "VisionTextDualEncoderConfig"}
```
Comment (ydshieh) [inline src/transformers/pipelines/__init__.py:418]: ok, only internal, good for me

File: tests/pipelines/test_pipelines_image_to_text.py [hunk 2]
```diff
@@ -257,14 +257,16 @@ def test_large_model_tf(self):
...
         prompt = (
             "<image>\nUSER: What does the label 15 represent? (1) lava (2) core (3) tunnel (4) ash cloud?\nASSISTANT:"
         )
 
-        outputs = pipe(image, prompt=prompt, generate_kwargs={"max_new_tokens": 200})
+        outputs = pipe(
+            "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/ai2d-demo.jpg",
+            prompt=prompt,
+            generate_kwargs={"max_new_tokens": 200},
+        )
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_image_to_text.py:269]: may I know why this new part is relevant for this PR?

**Review by ydshieh — APPROVED** [review]
Thanks a lot for this fix!

File: tests/pipelines/test_pipelines_image_to_text.py (hunk 2 above)
Comment (NielsRogge) [reply tests/pipelines/test_pipelines_image_to_text.py:269]: The more tests the better I assume :)

File: src/transformers/pipelines/__init__.py [hunk 3]
```diff
@@ -415,7 +415,8 @@
...
+MULTI_MODEL_SPEECH_CONFIGS = {"SpeechEncoderDecoderConfig"}
```
Comment (ArthurZucker) [inline src/transformers/pipelines/__init__.py:418]:
```suggestion
MULTI_MODEL_AUDIO_CONFIGS = {"SpeechEncoderDecoderConfig"}
```

**Review by ArthurZucker — APPROVED** [review]
Think it's a modality difference, fine for me. Could be `AUDIO_MODEL_CONFIGS` and VISION

File: src/transformers/pipelines/__init__.py (hunk 3 above)
Comment (amyeroberts) [inline src/transformers/pipelines/__init__.py:418]: Do we mean multi-modal here?

File: src/transformers/pipelines/__init__.py [hunk 4]
```diff
@@ -896,7 +897,10 @@ def pipeline(
         and not load_tokenizer
         and normalized_task not in NO_TOKENIZER_TASKS
         # Using class name to avoid importing the real class.
-        and model_config.__class__.__name__ in MULTI_MODEL_CONFIGS
+        and (
+            model_config.__class__.__name__ in MULTI_MODEL_SPEECH_CONFIGS
+            or model_config.__class__.__name__ in MULTI_MODEL_VISION_CONFIGS
```
Comment (amyeroberts) [inline src/transformers/pipelines/__init__.py:902]:
```suggestion
            model_config.__class__.__name__ in [MULTI_MODEL_SPEECH_CONFIGS, MULTI_MODEL_VISION_CONFIGS]
```

File: tests/pipelines/test_pipelines_image_to_text.py (hunk 2 above)
Comment (amyeroberts) [reply tests/pipelines/test_pipelines_image_to_text.py:259]: I'm also a little unclear - doesn't this mean this test was failing before as `outputs` wasn't defined?

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing! 

Just some small nits and comments

File: src/transformers/pipelines/__init__.py (hunk 3 above)
Comment (NielsRogge) [reply src/transformers/pipelines/__init__.py]: I'm not sure, I think it refers to multiple models (audio model and text model for instance).

**Pull request merged**